From 5723860d5035d974b2863a98c9d7cf6988a9ca40 Mon Sep 17 00:00:00 2001
From: Avinash Kumar <avikumar@habana.ai>
Date: Wed, 30 Jun 2021 14:18:35 +0300
Subject: [PATCH] [SW-0] gaudi2: add weak function for host_gpio

Gaudi2_arc_gnu preboot compilation fails if weak function
is in preboot.c and not moved to cmd_module.c For recovery.bin
creation preboot.c is not used hence compilation fails.

Change-Id: I0fb8f3c58f576fde25db854c27066632dea2573f
Signed-off-by: Avinash Kumar <avikumar@habana.ai>
---

diff --git a/common/cmd_module.c b/common/cmd_module.c
index a81db46..b5d84b2 100644
--- a/common/cmd_module.c
+++ b/common/cmd_module.c
@@ -10,6 +10,8 @@
 #include "uart.h"
 #include "cmd_module.h"
 
+__weak bool preboot_host_gpio_asserted(void) { return true; }
+
 static void i2c_debug_mode_handling(void)
 {
 	bool debug_mode;
diff --git a/preboot/preboot.c b/preboot/preboot.c
index f80aa9b..aa8210d 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -248,7 +248,6 @@
 __weak void preboot_led_on(void) {}
 __weak void preboot_sp_sram_init(void) {}
 __weak void hl_ram_init(void) {}
-__weak bool preboot_host_gpio_asserted(void) { return true; }
 
 static bool preboot_is_overlap_address_range(u64 addr, int len)
 {
